### PR TITLE
Implement SA1122 diagnostic and code fix

### LINKED ISSUE #55: SA1122: UseStringEmptyForEmptyStrings
http://www.stylecop.com/docs/SA1122.html

### PR DESCRIPTION
**Pull request by pdelvo**
Fixes #55

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1122UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    }
+}";
+
+            await VerifyCSharpFixAsync(string.Format(oldSource, useVerbatimLiteral ? "@" : ""), newSource);
+        }
+
+        [TestMethod]
+        public async Task TestCodeFixMultipleNodes()
+        {
+            // Tests if the code fix works if the SourceSpan of the diagnostic has more then one SynatxNode associated with it
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1122UnitTests.cs]: Nice, I wouldn't have thought of this.

**Comment by sharwell** [comment]
Can you add the following tests? I believe they would fail with the current code.

1. A string literal passed as the argument to an attribute.
2. A string literal which is assigned to a `const` field or local variable.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1122UnitTests.cs (hunk 1 above)
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1122UnitTests.cs]: I wouldnt have thought about this too if i would not have tested this with a small project I have with a lot of weird syntax in it ;)

**Comment by sharwell** [comment]
Thanks! By the way, make sure to comment after updating or it won't notify me. :+1:

**Pull request merged**